packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveMetadata/forceSourceRetrieveCmp.ts

```diff
@@ -66,15 +80,73 @@ export class ForceSourceRetrieveExecutor extends SfdxCommandletExecutor<
     });
     return quantities;
   }
+
+  public async execute(response: ContinueResponse<ApexDebugLogIdStartTime>): Promise<void> {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+    const execution = new CliCommandExecutor(this.build(response), {
+      cwd: getRootWorkspacePath()
+    }).execute(cancellationToken);
+    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+
+    const result = await new CommandOutput().getCmdResult(execution);
+    const resultJson = JSON.parse(result);
+    if (resultJson.status === 0 && this.openAfterRetrieve) {
+      const extensions = MetadataDictionary.getInfo(
+        resultJson.result.inboundFiles[0].type
+      )?.extensions;
+
+      resultJson.result.inboundFiles.map( async (item: any) => {
+        let fileToOpen;
+        if (extensions?.includes( '.' + item.filePath.split('.').slice(-1) )) {
```

Comment:
Consider using [path.extname](https://nodejs.org/api/path.html#path_path_extname_path) to pick out the extension

Reply:
Done!